src/Domain/Services/CheckoutFields.php

```diff
@@ -0,0 +1,653 @@
+<?php
+
+namespace Automattic\WooCommerce\Blocks\Domain\Services;
+
+use Automattic\WooCommerce\Blocks\Assets\AssetDataRegistry;
+use WC_Customer;
+
+/**
+ * Service class managing checkout fields and its related extensibility points.
+ */
+class CheckoutFields {
+
+
+	/**
+	 * Core checkout fields.
+	 *
+	 * @var array
+	 */
+	private $core_fields;
+
+	/**
+	 * Additional checkout fields.
+	 *
+	 * @var array
+	 */
+	private $additional_fields;
+
+	/**
+	 * Fields locations.
+	 *
+	 * @var array
+	 */
+	private $fields_locations;
+
+	/**
+	 * Instance of the asset data registry.
+	 *
+	 * @var AssetDataRegistry
+	 */
+	private $asset_data_registry;
+
+	/**
+	 * Billing fields meta key.
+	 *
+	 * @var string
+	 */
+	const BILLING_FIELDS_KEY = '_additional_billing_fields';
+
+	/**
+	 * Shipping fields meta key.
+	 *
+	 * @var string
+	 */
+	const SHIPPING_FIELDS_KEY = '_additional_shipping_fields';
+
+	/**
+	 * Additional fields meta key.
+	 *
+	 * @var string
+	 */
+	const ADDITIONAL_FIELDS_KEY = '_additional_fields';
+
+	/**
+	 * Sets up core fields.
+	 *
+	 * @param AssetDataRegistry $asset_data_registry Instance of the asset data registry.
+	 */
+	public function __construct( AssetDataRegistry $asset_data_registry ) {
+		$this->asset_data_registry = $asset_data_registry;
+		$this->core_fields         = array(
```

Comment:
Core fields have been moved here from JS. Note that this array exists else where in WooCommerce codebase, but it's filtered, we don't want to allow filtering this value yet.